flink-ml-lib/src/main/java/org/apache/flink/ml/common/datastream/TableUtils.java

```diff
@@ -38,4 +43,28 @@ public static RowTypeInfo getRowTypeInfo(ResolvedSchema schema) {
         }
         return new RowTypeInfo(types, names);
     }
+
+    /** Constructs a RowTypeInfo from the given schema. */
+    public static RowTypeInfo getRowTypeInfo(Schema schema) {
```

Comment:
It looks like this this method is used to process the output of `KnnModelData.getModelSchema`, which generates a `Schema` instead of `ResolvedSchema`.

Since we have all the information regarding the schema of `KnnModelData`, could we generate its schema as `ResolvedSchema` directly? This could allow us to simplify the PR by removing `RowTypeInfo getRowTypeInfo(Schema schema)`.

BTW, it is likely that other algorithms (i.e. `NaiveBayes`, `LogisticRegression` and `KMeans`) have already solved problems with other/simpler ways. Maybe we can learn from each other's PR and make the solution hopefully consistent.

Reply:
done